### PR TITLE
refactor(git-util):  extract GitProvider interface in its own file

### PR DESCRIPTION
**Pull request by tupe12334**
Partily: #5113

### CODE REVIEW

File: packages/amplication-git-utils/src/IGitProvider.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  GetPullRequestForBranchArgs,
+  GetRepositoriesArgs,
+  GetRepositoryArgs,
+  GitFile,
+  RemoteGitOrganization,
+  RemoteGitRepos,
+  RemoteGitRepository,
+} from "./types";
+
+export interface IGitProvider {
```
Comment (overbit) [inline packages/amplication-git-utils/src/IGitProvider.ts:18]: There is no convention to prefix interfaces in Ts with `I` and we are not currently using this convention elsewhere. Please rename it to `GitProvider`

**Review by overbit — CHANGES_REQUESTED** [review]

**Comment by abrl91** [comment]
@overbit @tupe12334 
This change doesn't conform to our conventions
[File & Declaration Naming](https://www.notion.so/amplication/File-Declaration-Naming-666afe407ff14ed6aa54bcd3dead45a8)

**Comment by overbit** [comment]
Good point @abrl91 . Please @tupe12334 rename this

**Comment by abrl91** [comment]
@overbit 
It's not only the renaming. All shared types/enums/interfaces should be in the same file. This interface is shared between all providers

**Comment by tupe12334** [comment]
I think we need to refactor the notion file because we don't work like that anywhere not even in our generated code

**Pull request merged**